trieste/bayesian_optimizer.py

```diff
@@ -669,11 +669,23 @@ def optimize(
 
                 with Timer() as total_step_wallclock_timer:
                     with Timer() as initial_model_fitting_timer:
-                        if step == 0 and fit_initial_model:
+                        if step == 1 and fit_initial_model:
                             for tag, model in models.items():
                                 dataset = datasets[tag]
                                 model.update(dataset)
                                 model.optimize(dataset)
+                            
+                    if step == 1 and fit_initial_model and summary_writer:
+                        logging.set_step_number(0)
+                        with summary_writer.as_default(step=0):
```

Comment:
@uri-granta for some reason this didn't work without first doing `logging.set_step_number(0)`, although if I understand correctly it should have worked without issues...

Reply:
The summary writer step argument is separate from the logging step state so you have to specify both. You can use a context manager if you prefer though, so you don't have to worry about resetting it afterwards.

```suggestion
                        with logging.step_number(0):
                            with summary_writer.as_default(step=0):
```

Reply:
thanks for the suggestion, that's definitely a better way to do it

Reply:
ok, that didnt work at the end for logging.scalar line below, logging wallclock model fitting time, so I reverted back

Reply:
Was the indentation correct? What error did you get? The context manager should be totally equivalent to setting and resetting like you're doing now, but cleaner and more robust.

Reply:
yes, indentation was correct I think, there was no error, scalar simply wasn't logged at all